src/main/java/de/tum/in/www1/artemis/versioning/VersionRangeComparator.java

```diff
@@ -0,0 +1,117 @@
+package de.tum.in.www1.artemis.versioning;
+
+import static de.tum.in.www1.artemis.versioning.VersionRangeComparisonType.*;
+import static de.tum.in.www1.artemis.versioning.VersionRangeService.versionRangeToIntegerList;
+
+import java.util.List;
+
+import de.tum.in.www1.artemis.exception.ApiVersionRangeNotValidException;
+
+/**
+ * Holds comparison logic for {@link VersionRange} annotations.
+ */
+public class VersionRangeComparator {
+
+    /**
+     * @param a Item containing two versions sorted ASC
+     * @param b Item to be compared to containing two versions sorted ASC
+     * @return A {@link VersionRangeComparisonType}
+     */
+    public static VersionRangeComparisonType compare(VersionRange a, VersionRange b) {
```

Comment:
I think some examples would be nice to quickly understand what all the version comparison classes output.

Reply:
I described each type in `VersionRangeComparionType`. Could you specify what additional documentation you would find helpful and where?

Reply:
What I understand so far is we are comparing ranges of version numbers? Is that correct?
Then maybe a few examples would help to understand the code quicker: `Comparing version ranges 1-5  and 2-6 returns xyz`.

Reply:
Do the changes work for you?

Reply:
Thanks that is great. 👍